Benchmarks/.gitignore

```diff
@@ -0,0 +1,9 @@
+.DS_Store
```

Comment:
Can we nest `Benchmarks` in `IntegrationTests` instead of the root of the repo?

Reply:
I intentionally did not nest it since I expect us to migrate all of them at some point and having it at the top level is way nicer. This also follows what we do in other packages.

Reply:
I don't think that's a good reason given `IntegrationTests` contains more than just allocation tests.

Reply:
I still would like to have them at the top level. Similar to how we should move the performance tests out of the `NIOPerformanceTester` and into `Benchmarks` once we migrate them. @Lukasa @dnadoba WDYT?

Reply:
I'm ok with having them at the top level.